operator/v1/types_metal3provisioning.go

```diff
@@ -0,0 +1,109 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Metal3Provisioning contains configuration used by the Provisioning
+// service (Ironic) to provision baremetal hosts.
+//
+// Metal3Provisioning is created by the Openshift installer using admin
+// or user provided information about the provisioning network and the NIC
+// on the server that can be used to PXE boot it.
+//
+// This CR is a singleton, created by the installer and currently only
+// consumed by the machine-api-operator to bring up and update containers
+// in a metal3 cluster.
+//
+type Metal3Provisioning struct {
```

Comment:
I don't think this is an appropriate name for this object.  `Metal3` is not a human facing thing - it's a project name.  This either belongs in an object named `Provisioning` in the `metal3.io` group, or should be human focused here `BareMetalProvisioning`

Reply:
When we run "oc get pods" we do see the metal3 -xxx-xx pod running. So, I assumed it would be OK to use/re-use metal3 here.